pkg/util/detector/detector_test.go

```diff
@@ -0,0 +1,51 @@
+package detector
+
+import (
+	"reflect"
+	"testing"
+
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util/informermanager/keys"
+)
+
+func TestResourceDetector_GetMatching(t *testing.T) {
+	type fields struct {
+		waitingObjects map[keys.ClusterWideKey]struct{}
+	}
+	type args struct {
+		resourceSelectors []policyv1alpha1.ResourceSelector
+	}
```

Comment:
Seems we don't need this type, just using `policyv1alpha1.ResourceSelector` would be fine.

Reply:
Done. Same as below.